File: acad-requisito-service/src/main/java/com/acad/requisitoservice/Entity/Requisito.java
package com.acad.requisitoservice.Entity;
import com.acad.requisitoservice.Dto.AntecedenteMedicoDto;
import com.acad.requisitoservice.Dto.ApoderadoDto;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "requisito")
public class Requisito {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_requisitos")
    private Long idRequisitos;

    @Column(name = "dni_estudiante", length = 20)
    private String dniEstudiante;

    @Column(name = "partida_nacimiento_original", length = 100)
    private String partidaNacimientoOriginal;

    @Column(name = "resolucion_rectoral_de_traslado", length = 100)
    private String resolucionRectoralDeTraslado;

    @Column(name = "certificado_de_estudio", length = 100)
    private String certificadoDeEstudio;

    @Column(name = "ficha_matricula_generado_por_siage", length = 100)
    private String fichaMatriculaGeneradoPorSIAGE;

    @Column(name = "constancia_de_comportamiento", length = 100)
    private String constanciaDeComportamiento;

    @Column(name = "constancia_de_no_adeudo", length = 100)
    private String constanciaDeNoAdeudo;

    @Column(name = "familiar_militar", length = 100)
    private String familiarMilitar;


    private Long idApoderado;

    @Transient

    private ApoderadoDto apoderado;


    private Long idAntecedenteMedico;

    @Transient

    private AntecedenteMedicoDto antecedenteMedico;

}

File: acad-front/src/app/category/requisito/requisito.ts
import { Component, OnInit } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { Router } from '@angular/router';
import {RequisitoModel} from '../models/requisito-model';
import {ApoderadoModel} from '../models/apoderado-model';
import {AntecedenteMedicoModel} from '../models/antedecedente-medico-model';
import {RequisitoService} from '../../core/service/requisito/requisito.service';
import {ApoderadoService} from '../../core/service/apoderado/apoderado.service';
import {AntecedenteMedicoService} from '../../core/service/antecedenteMedico/antecedente-medico.service';
import {NewRequisitoComponent} from './new/new-requisito.component';


@Component({
  selector: 'app-requisito',
  standalone: true,
  templateUrl: './requisito.html',
  styleUrl: './requisito.scss',
  imports: [CommonModule, FormsModule, NewRequisitoComponent],
})
export class Requisito implements OnInit {

  requisitos: RequisitoModel[] = [];
  apoderados: ApoderadoModel[] = [];
  antecedentesMedicos: AntecedenteMedicoModel[] = [];

  nuevoRequisitoVisible = false;

  nuevoRequisito: RequisitoModel = {
    dniEstudiante: '',
    partidaNacimientoOriginal: '',
    resolucionRectoralDeTraslado: '',
    certificadoDeEstudio: '',
    fichaMatriculaGeneradoPorSIAGE: '',
    constanciaDeComportamiento: '',
    constanciaDeNoAdeudo: '',
    familiarMilitar: '',
    idApoderado: undefined,
    idAntecedenteMedico: undefined
  };

  constructor(
    private requisitoService: RequisitoService,
    private apoderadoService: ApoderadoService,
    private antecedenteService: AntecedenteMedicoService,
    private router: Router
  ) {}

  ngOnInit(): void {
    this.cargarRequisitos();
    this.cargarApoderados();
    this.cargarAntecedentesMedicos();
  }

  cargarRequisitos(): void {
    this.requisitoService.getRequisitos().subscribe(data => {
      this.requisitos = data;
    });
  }

  cargarApoderados(): void {
    this.apoderadoService.getApoderados().subscribe(data => {
      this.apoderados = data;
    });
  }

  cargarAntecedentesMedicos(): void {
    this.antecedenteService.getAntecedenteMedico().subscribe(data => {
      this.antecedentesMedicos = data;
    });
  }

  onNuevoRequisitoGuardado(): void {
    this.nuevoRequisitoVisible = false;
    this.cargarRequisitos();
  }

  mostrarFormularioNuevo(): void {
    this.nuevoRequisitoVisible = !this.nuevoRequisitoVisible;
  }

  editar(id?: number): void {
    if (id != null) {
      this.router.navigate(['/requisitos/editar', id]);
    }
  }

  eliminar(id?: number): void {
    if (id == null) return;

    if (confirm('¿Estás seguro de eliminar este requisito?')) {
      this.requisitoService.delete$(id).subscribe(() => {
        this.requisitos = this.requisitos.filter(r => r.idRequisitos !== id);
      });
    }
  }
}
